crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -73,36 +70,51 @@ impl Violation for Airflow3Removal {
     }
 }
 
-fn diagnostic_for_argument(
-    arguments: &Arguments,
-    deprecated: &str,
-    replacement: Option<&'static str>,
-) -> Option<Diagnostic> {
-    let keyword = arguments.find_keyword(deprecated)?;
-    let mut diagnostic = Diagnostic::new(
-        Airflow3Removal {
-            deprecated: (*deprecated).to_string(),
-            replacement: match replacement {
-                Some(name) => Replacement::Name(name),
-                None => Replacement::None,
-            },
-        },
-        keyword
-            .arg
-            .as_ref()
-            .map_or_else(|| keyword.range(), Ranged::range),
-    );
+/// AIR302
+pub(crate) fn removed_in_3(checker: &mut Checker, expr: &Expr) {
+    if !checker.semantic().seen_module(Modules::AIRFLOW) {
+        return;
+    }
 
-    if let Some(replacement) = replacement {
-        diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
-            replacement.to_string(),
-            diagnostic.range,
-        )));
+    match expr {
+        Expr::Call(ExprCall {
+            func, arguments, ..
+        }) => {
+            if let Some(qualname) = checker.semantic().resolve_qualified_name(func) {
+                removed_argument(checker, &qualname, arguments);
+            };
+
+            removed_method(checker, expr);
+        }
+        Expr::Attribute(ExprAttribute { attr: ranged, .. }) => {
+            removed_name(checker, expr, ranged);
+            removed_class_attribute(checker, expr);
+        }
+        ranged @ Expr::Name(ExprName { id, ctx, .. }) => {
+            removed_name(checker, expr, ranged);
+            if ctx == &ExprContext::Store {
+                if let ScopeKind::Class(class_def) = &checker.semantic().current_scope().kind {
+                    removed_airflow_plugin_extension(checker, expr, id, class_def);
+                }
+            }
+        }
+        _ => {}
     }
+}
 
-    Some(diagnostic)
+#[derive(Debug, Eq, PartialEq)]
+enum Replacement {
+    None,
+    Name(&'static str),
+    Message(&'static str),
 }
 
+// Check whether a removed Airflow argument is passed.
```

Comment:
In Rust, documentation comments starts with triple forward slashes (`///`). I'll update this in my planned follow-up PR.

Reply:
Got it!